protostar/testing/environments/fuzz_test_execution_environment.py

```diff
@@ -127,6 +133,18 @@ def build_and_run_test(
         given_strategies: Dict[str, SearchStrategy],
     ):
         try:
+            phases = (
+                Phase.explicit,
+                Phase.reuse,
+                Phase.generate,
+                Phase.target,
+                Phase.shrink,
+            )
```

Comment:
```suggestion
            phases = tuple(Phase)
```

Reply:
the default value for this is actually `set(Phase) - {Phase.explain}`

Reply:
[simplify phases](https://github.com/software-mansion/protostar/pull/947/commits/3882f41e145d8acf1f71d031b4159c18b0a163e1)

Reply:
Neither of these approaches turned out to be completely clean.
- `settings.default` makes `pyright` complain, just like using `not_set` (different error being yield ofc)
- I could not really operate on the `settings object`, I wanted to use the default value for the `phases` and modify it conditionally. Unfortunately: `AttributeError: settings objects are immutable`.

I think the best way to go here is to just use `# type: ignore` and move on.

Reply:
[use default phases](https://github.com/software-mansion/protostar/pull/947/commits/8e0e9577459959a5e70c83373ca172e84a07ed29)

Reply:
as discussed privately, there's a way to "mutate" settings, let's do that here